plugins/module_utils/ansible_freeipa_module.py

```diff
@@ -179,19 +179,26 @@ def api_connect(context=None):
 
         `context` can be any of:
             * `server` (default)
-            * `ansible-freeipa`
-            * `cli_installer`
+            * `client`
         """
         env = Env()
         env._bootstrap()
         env._finalize_core(**dict(DEFAULT_CONFIG))
 
-        # available contexts are 'server', 'ansible-freeipa' and
-        # 'cli_installer'
-
+        # Default context is 'server'.
         if context is None:
```

Comment:
How about using a client context on a client automatically if the context is not set? This way a task could be executed on a client without specifying the context.

Reply:
Otherwise context would be a required parameter on clients always.

Reply:
if context is None:
        if api.env.in_server:
            context = "server"
        else:
            context = "client"

Reply:
I was thinking on forcing explicit usage if client context is to be used. Mostly because I didn't know if all modules would run on it. I'll change it to select the proper context, and change documentation, as `ipaapi_context` will not default to `server`.

Reply:
fixed.